pkg/yttlibrary/url.go

```diff
@@ -178,3 +185,59 @@ func (b urlModule) sortedKeys(vals url.Values) []string {
 	sort.Strings(result)
 	return result
 }
+
+func (b urlModule) ParseURL(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 1 {
+		return starlark.None, fmt.Errorf("expected exactly one argument")
+	}
+
+	urlStr, err := core.NewStarlarkValue(args.Index(0)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	return (&urlValue{parsedURL}).AsStarlarkValue(), nil
+}
+
+func (uv *urlValue) AsStarlarkValue() starlark.Value {
+	return &starlarkstruct.Module{
```

Comment:
This value feels to me more like a `starlarkstruct.Struct` than a `starlarkstruct.Module`: `urlValue` _is_ a data structure.

I do see that `libraryValue` has been implemented as a `module`...
https://github.com/vmware-tanzu/carvel-ytt/blob/bc8782e60d1089a4781060d36811fea5bbbb608b/pkg/workspace/library_module.go#L143-L153

Yet, while it's all methods, it too is a value, not a result of an evaluated Starlark Module.

That `TODO` on line 143 is legit: this isn't a pattern we want to propagate.

...

There's a "constructor" for `Structs`:
https://github.com/vmware-tanzu/carvel-ytt/blob/bc8782e60d1089a4781060d36811fea5bbbb608b/vendor/github.com/k14s/starlark-go/starlarkstruct/struct.go#L50-L52

And `Builtin`s (i.e. a function implemented in Go) can be bound to any Starlark value:
https://github.com/vmware-tanzu/carvel-ytt/blob/bc8782e60d1089a4781060d36811fea5bbbb608b/vendor/github.com/k14s/starlark-go/starlark/value.go#L663-L676